src/process/dust/ccpr_scheme_fengsha_mod.F90

```diff
@@ -163,6 +163,10 @@ subroutine CCPr_Scheme_Fengsha(nDustSpecies,    &
 
       nbins = size(reff)
 
+      if (present(MoistOpt) .eqv. .false.) MoistOpt = 1
+      if (present(DragOpt) .eqv. .false.) DragOpt = 1
+      if (present(HorizFluxOpt) .eqv. .false.) HorizFluxOpt = 1
```

Comment:
Unfortunately, you really need to have a separate variable for this to work properly in both cases.

```
integer, intent(in), optional :: MoistOpt  ! optional input

integer :: MoistOpt_  ! local variable to actually use

MoistOpt_ = 1  ! default
if (present(MoistOpt)) MoistOpt_ = MoistOpt  ! override default if provided

! ...
! Use `MoistOpt_` in the code
```